### PR TITLE
feat: search results type from index name

### PR DESCRIPTION
**Pull request by yassilah**
from: https://github.com/nuxt-community/algolia-module/discussions/24

### CODE REVIEW

**Comment by Baroshem** [comment]
Could you please add also a mention in the docs how a user can use it? Playground is used when you download the repo and try it locally. For package users will search through our docs instead :)

**Comment by Baroshem** [comment]
@yassilah are you there? ;)

**Comment by yassilah** [comment]
Hey! I was on vacation and didn't bring my computer to disconnect but i'll get on it now 🙂

File: docs/pages/4.advanced.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 title: Advanced
-description: 'Learn more about the algolia module for your Nuxt application.'
+description: "Learn more about the algolia module for your Nuxt application."
```
Comment (Baroshem) [inline docs/pages/4.advanced.md:3]: I think you have linter applying changes to the documentation projects as well that can result in some issues. Could you please revert these changes?

File: docs/pages/4.advanced.md [hunk 2]
```diff
@@ -17,28 +17,25 @@ By default, Algolia module only provides the search functionality but you can en
...
+});
 ```
 
 Next, let's create `indexName` variable and call `useAlgolia` composable to get the reference to Algolia:
 
 ```ts
 <script lang="ts" setup>
-const indexName = 'test_index'
-const algolia = useAlgolia()
+  const indexName = 'test_index' const algolia = useAlgolia()
```
Comment (Baroshem) [inline docs/pages/4.advanced.md:38]: Issues like this one here

File: docs/pages/4.advanced.md [hunk 3]
```diff
@@ -54,3 +51,33 @@ Finally, let's use it in our page.vue with vue-instantsearch components:
...
+  }
+}
+```
+
+You may then use your Algolia indices and enjoy typed search hits 🔥
+
+```vue
+<script lang="ts" setup>
+const { search } = useInitIndex("someIndex"); // Auto-completion for index names
+const { hits } = await search("my query");
```
Comment (Baroshem) [inline docs/pages/4.advanced.md:78]: For search you need to have an object like following:
```ts
await search({ query: "my query" })
```

**Review by Baroshem — CHANGES_REQUESTED** [review]
Some small changes needed but apart from that, very good :)

File: docs/pages/4.advanced.md (hunk 1 above)
Comment (yassilah) [reply docs/pages/4.advanced.md]: Oops yes sorry about that

File: docs/pages/4.advanced.md (hunk 3 above)
Comment (yassilah) [reply docs/pages/4.advanced.md]: I just checked and when using "useInitIndex", the query is just a string; it's when using the `useSearch` composable that the query is an object. But I also figured it would make sense to add the same typings for the other composables, so I added them for `useSearch` and `useSearchForFacetValues`.

Comment (Baroshem) [reply docs/pages/4.advanced.md]: Yes, because useInitIndex does not have any methods. It only accepts an index name and returns a Algolia index out of it. And it is not a query, but an indexName ;) 

Thanks for adding that in other places as well!

File: docs/pages/4.advanced.md [hunk 4]
```diff
@@ -54,3 +52,44 @@ Finally, let's use it in our page.vue with vue-instantsearch components:
...
+    }
+  }
+}
+```
+
+You may then use your Algolia indices and enjoy typed search hits 🔥
+
+```vue
+<script lang="ts" setup>
+const { search } = useInitIndex('someIndex') // Auto-completion for index names
```
Comment (Baroshem) [inline docs/pages/4.advanced.md:78]: Ok, I see what you did here :D 

useInitIndex should be used mainly to have a reference to the index itself (similarly to the useAlgolia, we use it to have a reference to algoliasearch). 

In this case I would only leave the example with the useSearch as this should be used in order to handle search functionality using Algolia.

So something like this:

```ts
const { result, search } = useSearch('someIndex')

await search({ query: 'test'  }) 
```

useSearch uses useInitIndex under the hood btw.

**Review by Baroshem — APPROVED** [review]
Let's wait for the discussion creator to also approve and then we could merge it :)

**Review by Rigo-m — APPROVED** [review]
I think this is it. Super cool feature imho, great job guys!
